Mayhem

capuanob opened the pull request:
This includes my contributions for harnessing Dimsum with Mayhem. Please let me know if anything needs to be adjusted!

capuanob committed:
Added GitHub Action to launch Mayhem

capuanob committed:
Added CMakeList and modifications to fuzzer to branch

capuanob committed:
Bug fix: CMakeList was accidentally reset

xansec reviewed (changes requested):
I don't believe using CMake here is necessary, it looks like there is already a build system to build the fuzz target. Please use this build system instead, and remove the CMakeLists.txt (https://github.com/google/dimsum/blob/master/BUILD.bazel#L53)

capuanob committed:
PR Fix: Using Bazel build system for Mayhem Integration

capuanob commented:
Alright, I've gone ahead and updated the Dockerfile to utilize the existing Bazel build system and also cleaned up my packaging a tad bit. Please let me know if any further changes are necessary!

xansec commented on WORKSPACE:
```diff
@@ -1,7 +1,8 @@
 http_archive(
      name = "com_google_googletest",
      urls = ["https://github.com/google/googletest/archive/master.zip"],
-     strip_prefix = "googletest-master",
```
One final question - can you comment on this change here?

capuanob replied:
Two changes had to be made to this file for a successful build.
1. Since Bazel 0.20, use of native http_archive has been deprecated. Ref: [here](https://stackoverflow.com/questions/61163675/error-http-archive-is-not-defined-when-starting-the-local-bazel-server-in-the)
2. Since last update to the bazel config, the remote has been renamed to googletest-main.

xansec commented:
Gotcha, LGTM!
